### PR TITLE
(feat): 7 Day Feedback survey

### PR DESCRIPTION
**Pull request by laurieontech**
We should ask for specific user feedback after their first 7 days.
[ch17397]

### CODE REVIEW

**Comment by boborchard** [comment]
@laurieontech The survey link will be https://www.surveymonkey.com/r/gatsbyjs-feedback-cli

File: packages/gatsby/src/utils/feedback.ts [hunk 1]
```diff
@@ -111,3 +107,39 @@ export async function userPassesFeedbackRequestHeuristic(): Promise<boolean> {
...
 }
+
+function isFeedbackDisabled(): boolean {
+  // Heuristic 2
+  if (getConfigStore().get(feedbackKey) === true) {
+    return true
+  }
+
+  // Heuristic 3
+  if (process.env.GATSBY_FEEDBACK_DISABLED === `1`) {
```
Comment (jamo) [inline packages/gatsby/src/utils/feedback.ts:118]: We have a convenience method for these: https://github.com/gatsbyjs/gatsby/blob/ecc0ad37f3fb6d542964c95c2f853c25cc99d5b8/packages/gatsby-telemetry/src/is-truthy.ts#L3
not sure if it's worth moving to core or elsewhere.

Comment (laurieontech) [reply packages/gatsby/src/utils/feedback.ts:144]: Noted. I think in this one off case we're ok, but certainly something to be aware of moving forward.

Comment (jamo) [reply packages/gatsby/src/utils/feedback.ts:144]: 👍

File: packages/gatsby/src/utils/feedback.ts [hunk 2]
```diff
@@ -22,6 +25,19 @@ export function showFeedbackRequest(): void {
   report.log(`\nGive us your feedback here: https://gatsby.dev/feedback\n\n`)
 }
 
+export function showSevenDayFeedbackRequest(): void {
+  getConfigStore().set(sevenDayKey, Date.now())
+  trackCli(`SHOW_SEVEN_DAY_FEEDBACK_LINK`, {
+    name: `https://www.surveymonkey.com/r/gatsbyjs-feedback-cli`,
+  })
+  report.log(
+    `\n\nHello! Will you help Gatsby improve by taking a four question survey?\nIt takes less than five minutes and your ideas and feedback will be very helpful.`
```
Comment (boborchard) [inline packages/gatsby/src/utils/feedback.ts:34]: Let's give users an idea of what we're about to ask in the survey:

> \n\nHi there! Will you tell us about how you're learning Gatsby? \nIt takes less than 5 minutes and your feedback will help us make installing and using Gatsby so much better.

File: packages/gatsby/src/utils/feedback.ts [hunk 3]
```diff
@@ -22,6 +25,19 @@ export function showFeedbackRequest(): void {
   report.log(`\nGive us your feedback here: https://gatsby.dev/feedback\n\n`)
 }
 
+export function showSevenDayFeedbackRequest(): void {
+  getConfigStore().set(sevenDayKey, Date.now())
+  trackCli(`SHOW_SEVEN_DAY_FEEDBACK_LINK`, {
+    name: `https://www.surveymonkey.com/r/gatsbyjs-feedback-cli`,
```
Comment (boborchard) [inline packages/gatsby/src/utils/feedback.ts]: Let's use a link similar to the other survey: `https://gatsby.dev/feedback-survey`

**Review by boborchard — CHANGES_REQUESTED** [review]

File: packages/gatsby/src/utils/feedback.ts (hunk 3 above)
Comment (laurieontech) [reply packages/gatsby/src/utils/feedback.ts]: Already in there!

File: packages/gatsby/src/utils/feedback.ts [hunk 4]
```diff
@@ -111,3 +133,38 @@ export async function userPassesFeedbackRequestHeuristic(): Promise<boolean> {
...
+
+  // Heuristic 3
+  if (process.env.GATSBY_FEEDBACK_DISABLED === `1`) {
+    return true
+  }
+  return false
+}
+
+export async function userGetsSevenDayFeedback(): Promise<boolean> {
+  if (isCI || isFeedbackDisabled()) return false
```
Comment (wardpeet) [inline packages/gatsby/src/utils/feedback.ts:151]: I'm not 100% sure but isn't isCI a function?
```suggestion
  if (isCI() || isFeedbackDisabled()) return false
```

Comment (wardpeet) [reply packages/gatsby/src/utils/feedback.ts:151]: Shouldn't we add isCi() to the isFeedbackDisabled call instead?

Comment (laurieontech) [reply packages/gatsby/src/utils/feedback.ts]: We certainly can. That's shared between both calls and right now (for some reason) we don't check CI for the regular feedback request. If we're good to change that behavior then I'll add it.

File: packages/gatsby/src/utils/feedback.ts [hunk 5]
```diff
@@ -111,3 +133,43 @@ export async function userPassesFeedbackRequestHeuristic(): Promise<boolean> {
...
+  if (getConfigStore().get(feedbackKey) === true) {
+    return true
+  }
+
+  // Heuristic 3
+  if (process.env.GATSBY_FEEDBACK_DISABLED === `1`) {
+    return true
+  }
+
+  if (isCI) {
```
Comment (laurieontech) [inline packages/gatsby/src/utils/feedback.ts:148]:
```suggestion
  if (isCI()) {
```

**Review by pvdz — APPROVED** [review]

**Review by pvdz — APPROVED** [review]

**Review by ascorbic — APPROVED** [review]

**Pull request merged**